source-code/core/src/lint/report.ts

```diff
@@ -0,0 +1,19 @@
+import type { LintableNode, LintReport, LintRule, LintedNode } from "./rule.js"
+
+/**
+ * Creates a function that can be used to report lint issues.
+ */
+export const createReportFunction = (_args: { id: LintRule["id"]; level: LintRule["level"] }) => {
+	return (args: { node: LintableNode; message: string }) => {
+		if (!args.node) return
+
+		args.node.lint = [
+			...((args.node as LintedNode).lint ?? []),
+			{
+				id: _args.id,
+				level: _args.level,
+				...args,
```

Comment:
I would not add the `node` to this object. Just the message.

Reply:
good catch. i haven't changed this code though ;)